otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py

```diff
@@ -182,70 +114,50 @@ def _delete_instance(self, instance, as_group):
         )
 
     def _initialize_as_group_with_instance(self):
-        self.key_pair = self._create_keypair()
-        self.sec_group = self._create_sec_group()
-        self.network = self._create_network()
-        self.subnet = self._create_subnet(self.network.id)
-        self.router = self._create_router(self.subnet.id)
         self.as_config = self._create_as_config(
-            self._get_image_id(), self.sec_group.id
+            self._get_image_id(), self.infra.get("sec_group_id")
         )
         self.as_group = self._create_as_group(
-            self.as_config.id, self.router['id'], self.network.id
+            self.as_config.id, self.infra.get("router_id"),
+            self.infra.get("network_id")
         )
-        self.instance = self._wait_for_instance(
+        self.as_instance = self._wait_for_instance(
             self.as_group
         )
 
     def _deinitialize_as_group_with_instance(self):
-        if self.instance:
-            self._delete_instance(self.instance, self.as_group)
+        if self.as_instance:
+            self._delete_instance(self.as_instance)
         if self.as_group:
             self._delete_as_group(self.as_group)
         if self.as_config:
             self._delete_as_config(self.as_config)
-        if self.router:
-            self._delete_router(self.router, self.subnet.id)
-        if self.subnet:
-            self._delete_subnet(self.subnet)
-        if self.network:
-            self._delete_network(self.network)
-        if self.sec_group:
-            self._delete_sec_group(self.sec_group)
-        if self.key_pair:
-            self._delete_keypair(self.key_pair)
 
     def setUp(self):
-        test_timeout = 3 * int(os.environ.get('OS_TEST_TIMEOUT'))
-        try:
-            self.useFixture(
-                fixtures.EnvironmentVariable(
-                    'OS_TEST_TIMEOUT', str(test_timeout)))
-        except ValueError:
-            pass
         super(TestInstance, self).setUp()
         self._initialize_as_group_with_instance()
 
     def tearDown(self):
-        super(TestInstance, self).tearDown()
         try:
             self._deinitialize_as_group_with_instance()
-        except openstack.exceptions.SDKException as e:
+        except exceptions.SDKException as e:
             _logger.warning('Got exception during clearing resources %s'
                             % e.message)
+        finally:
```

Comment:
using finally is absolutely not required. Moving parent tearDown to the end makes sense, but finally is not required

Reply:
Done